### ISSUE TITLE
nn.AdaptiveMaxPool3d does not support tuple composed by None

### ISSUE THREAD

**Issue by Rhythmblue** [pytorch/pytorch#8789]
## Issue description
Doc of  [torch.nn.AdaptiveMaxPool3d](https://pytorch.org/docs/stable/nn.html#adaptivemaxpool3d):
output_size – the target output size of the image of the form D x H x W. Can be a tuple (D, H, W) or a single D for a cube D x D x D. D, H and W can be either a int, or None which means the size will be the same as that of the input.

The option of (7, None, None) cannot work on 0.4.0 now. It can work on 0.3.0.
I don't know how to fix it. Maybe a simple check statement can solve it.

## Code

```python
import torch
import torch.nn as nn

m = nn.AdaptiveMaxPool3d((7, None, None))
input = torch.randn(1, 64, 10, 9, 8)
output = m(input)
```

## Result
```
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    output = m(input)
  File "/home/zhouhao/lib_my/miniconda3/envs/pytorch4/lib/python3.6/site-packages/torch/nn/modules/module.py", line 491, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/zhouhao/lib_my/miniconda3/envs/pytorch4/lib/python3.6/site-packages/torch/nn/modules/pooling.py", line 885, in forward
    return F.adaptive_max_pool3d(input, self.output_size, self.return_indices)
  File "/home/zhouhao/lib_my/miniconda3/envs/pytorch4/lib/python3.6/site-packages/torch/nn/functional.py", line 506, in adaptive_max_pool3d
    ret = torch._C._nn.adaptive_max_pool3d(input, output_size)
TypeError: an integer is required (got type NoneType)

```

## System Info
PyTorch version: 0.4.0
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.3 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.5) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration:
GPU 0: GeForce GTX 1080 Ti
GPU 1: GeForce GTX 1080 Ti
GPU 2: GeForce GTX 1080 Ti
GPU 3: GeForce GTX 1080 Ti

Nvidia driver version: 384.111
cuDNN version: Probably one of the following:
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.6.0.21
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn_static.a
/usr/local/matlab/2018a/bin/glnxa64/libcudnn.so.7.0.3

Versions of relevant libraries:
[pip3] numpy (1.13.3)
[conda] cuda90                    1.0                  h6433d27_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch
[conda] pytorch                   0.4.0           py36_cuda9.0.176_cudnn7.1.2_1  [cuda90]  https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch
[conda] torchvision               0.2.1                    py36_0    https://mirrors.ustc.edu.cn/anaconda/cloud/conda-forge

**Comment by SsnL** [comment]
Fixing at #8596 !

**Comment by SsnL** [comment]
Fixed via #8596
